servicetalk-log4j2-mdc-utils/build.gradle

```diff
@@ -17,14 +17,15 @@
 apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-library"
 
 dependencies {
+  api platform(project(":servicetalk-dependencies"))
   testImplementation enforcedPlatform("org.junit:junit-bom:$junit5Version")
 
-  api "org.apache.logging.log4j:log4j-core:$log4jVersion"
+  api "org.apache.logging.log4j:log4j-core"
```

Comment:
Let me continue https://github.com/apple/servicetalk/pull/2210#discussion_r870547622 discussion here:

> `servicetalk-log4j2-mdc-utils` requires `log4j-core` for SPI types it exposes. I have tweaked the dependencies it uses and the dependencies referenced in the `-dependencies` BOM.

can you please clarify what requires `log4j-core`? I see only `log4j-api` classes in `servicetalk-log4j2-mdc-utils`

Reply:
I have tuned what is required more closely in the updated PR. Only the minimum necessary should be expressed now.